src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ControllerSaveTempDataPropertyFilter.cs

```diff
@@ -19,10 +25,29 @@ public void OnActionExecuted(ActionExecutedContext context)
         /// <inheritdoc />
         public void OnActionExecuting(ActionExecutingContext context)
         {
-            Subject = context.Controller;
             var tempData = _factory.GetTempData(context.HttpContext);
+            var viewData = context.HttpContext.Items[ViewDataDictionaryControllerPropertyActivator.ViewDataKey] as ViewDataDictionary ??
```

Comment:
This should really be its own system or needs to be generalized more. Insert quote about numbers that matter in computer science. @ajaybhargavb probably can tell you